solutions/fully-configurable/main.tf

```diff
@@ -0,0 +1,212 @@
+      region_location                     = var.region
+      resource_instance_id                = var.existing_cos_instance_crn
+      management_endpoint_type_for_bucket = var.management_endpoint_type_for_bucket
+      skip_iam_authorization_policy       = true
+    }
+  ]
+}
+
+data "ibm_cos_bucket" "existing_logs_bucket" {
+  count                = local.create_buckets ? 0 : 1
+  bucket_name          = var.existing_logs_cos_bucket_name
+  resource_instance_id = var.existing_cos_instance_crn
+  bucket_region        = var.existing_logs_cos_bucket_region
+  bucket_type          = var.existing_logs_cos_bucket_type
+}
+
+data "ibm_cos_bucket" "existing_metrics_bucket" {
+  count                = local.create_buckets ? 0 : 1
+  bucket_name          = var.existing_metrics_cos_bucket_name
+  resource_instance_id = var.existing_cos_instance_crn
+  bucket_region        = var.existing_metrics_cos_bucket_region
+  bucket_type          = var.existing_metrics_cos_bucket_type
+}
+
+module "existing_kms_crn_parser" {
+  count   = var.existing_kms_instance_crn != null ? 1 : 0
+  source  = "terraform-ibm-modules/common-utilities/ibm//modules/crn-parser"
+  version = "1.1.0"
+  crn     = var.existing_kms_instance_crn
+}
+
+module "existing_kms_key_crn_parser" {
+  count   = local.use_kms_module ? 1 : 0
+  source  = "terraform-ibm-modules/common-utilities/ibm//modules/crn-parser"
+  version = "1.1.0"
+  crn     = local.kms_key_crn
+}
+
+module "kms" {
+  count                       = local.use_kms_module ? 1 : 0
+  source                      = "terraform-ibm-modules/kms-all-inclusive/ibm"
+  version                     = "4.20.0"
+  create_key_protect_instance = false
+  region                      = var.region
+  existing_kms_instance_crn   = var.existing_kms_instance_crn
+  key_ring_endpoint_type      = var.kms_endpoint_type
+  key_endpoint_type           = var.kms_endpoint_type
+  keys = [
+    {
+      key_ring_name     = local.key_ring_name
+      existing_key_ring = false
+      keys = [
+        {
+          key_name                 = local.key_name
+          standard_key             = false
+          rotation_interval_month  = 3
+          dual_auth_delete_enabled = false
+          force_delete             = true
+        }
+      ]
+    }
+  ]
+}
+
+#######################################################################################################################
+# Event Notifications
+#######################################################################################################################
+
+locals {
+  # tflint-ignore: terraform_unused_declarations
+  pass_emails = length(var.existing_event_notifications_instances) > 0 && length(var.event_notifications_email_list) == 0 ? tobool("You must pass at least one email address if setting up Event Notifications Integration.") : true
+
+  en_topic              = try("${local.prefix}-Cloud Logs Topic", "Cloud Logs Topic")
```

Comment:
I think the topic name should have the Clpud logs GUID in it. Same way I added here for SCC: https://github.com/terraform-ibm-modules/terraform-ibm-scc/blob/2f888db2f038d0db6cffbb228cff53fd53652f04/solutions/fully-configurable/main.tf#L295